public/js/src/module/records-queue.js

```diff
@@ -124,6 +122,27 @@ function getAutoSavedRecord() {
     return get(getAutoSavedKey());
 }
 
+/**
+ * Updates the disabled/enabled state of the upload button.
+ */
+function _updateUploadButtonState() {
```

Comment:
Enabling and disabling the upload button was messy, as it was directly done in two different functions. And it appeared this resulted in a yet-undiscovered bug as well (when offline and adding an additional "final" record to the queue when the button was disabled, the upload button was enabled again until the next call to uploadQueue). 

To address issue #401, we'd be possibly adding a third place, increasing the chance of bugs even further.

I therefore chose to use a new function that always sets the button correctly no matter how many times and where it is called. It now is just a matter of calling it at the right times (at least).

It still relies on the semi-global `finalRecordPresent` and `uploadOngoing` variables. I considered getting rid of those, but failed to find a better solution and decided to consider it out of scope.